src/Avalonia.Input/GestureRecognizers/ScrollGestureRecognizer.cs

```diff
@@ -101,7 +101,7 @@ public void PointerMoved(PointerEventArgs e)
                 if (_scrolling)
                 {
                     var vector = _trackedRootPoint - rootPoint;
-                    var elapsed = _lastMoveTimestamp.HasValue ?
+                    var elapsed = _lastMoveTimestamp.HasValue && _lastMoveTimestamp < e.Timestamp ?
```

Comment:
Should this file also be updated on line 69?
https://github.com/AvaloniaUI/Avalonia/pull/7412/files#diff-7166ab0bdccc56e4e1ac95ad7c0dcca6e9c18d5c458df5f2f1acec5cdd35e373R69
So Pen scrolling will also work.

Reply:
How exactly can we get _lastMoveTimestamp to be greater that the current event timestamp? Are events comping out of order or something?

Reply:
Yes.